### PR TITLE
Ctctraders 1333: Confirm remove container page

### PR DESCRIPTION
**Pull request by kevincoatsworth**

### CODE REVIEW

File: app/pages/addItems/containers/ConfirmRemoveContainerPage.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (garethgooch) [inline app/pages/addItems/containers/ConfirmRemoveContainerPage.scala:1]: this should be a page as we never save this data

File: app/controllers/addItems/containers/ConfirmRemoveContainerController.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                "radios"         -> Radios.yesNo(formWithErrors("value"))
+              )
+
+              renderer.render(template, json).map(BadRequest(_))
+            },
+            value =>
+              if (value) {
+                for {
+                  updatedAnswers <- Future.fromTry(request.userAnswers.remove(ContainersQuery(index, containerIndex)))
+                  _              <- sessionRepository.set(updatedAnswers)
```
Comment (garethgooch) [inline app/controllers/addItems/containers/ConfirmRemoveContainerController.scala:98]: this should be ContainersNumberPage(itemIndex, contanerIndex)?

File: app/queries/ContainersQuery.scala [hunk 3]
```diff
@@ -21,7 +21,7 @@ import pages.QuestionPage
 import play.api.libs.json.{JsObject, JsPath}
 import queries.Constants.{containers, items}
```
Comment (garethgooch) [inline app/queries/ContainersQuery.scala:23]: containers query is used to pull back all containers

File: app/utils/AddItemsCheckYourAnswersHelper.scala [hunk 4]
```diff
@@ -585,6 +587,21 @@ class AddItemsCheckYourAnswersHelper(userAnswers: UserAnswers) {
       )
   }
 
+  def confirmRemoveContainer(index: Index, containerIndex: Index): Option[Row] = userAnswers.get(ConfirmRemoveContainerPage(index, containerIndex)) map {
```
Comment (garethgooch) [inline app/utils/AddItemsCheckYourAnswersHelper.scala:590]: Not required as unsaved and never displayed

File: test/pages/addItems/containers/ConfirmRemoveContainerPageSpec.scala [hunk 5]
```diff
@@ -0,0 +1,32 @@
+/*
```
Comment (garethgooch) [inline test/pages/addItems/containers/ConfirmRemoveContainerPageSpec.scala:1]: not required as just a page

File: test/generators/UserAnswersEntryGenerators.scala [hunk 6]
```diff
@@ -33,6 +33,13 @@ trait UserAnswersEntryGenerators {
 
   self: Generators =>
 
+  implicit lazy val arbitraryConfirmRemoveContainerUserAnswersEntry: Arbitrary[(ConfirmRemoveContainerPage, JsValue)] =
```
Comment (garethgooch) [inline test/generators/UserAnswersEntryGenerators.scala:36]: not required as not on UA

File: app/controllers/addItems/containers/ConfirmRemoveContainerController.scala [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    with I18nSupport
+    with NunjucksSupport {
+
+  private val form     = formProvider()
+  private val template = "addItems/containers/confirmRemoveContainer.njk"
+
+  def onPageLoad(lrn: LocalReferenceNumber, index: Index, containerIndex: Index, mode: Mode): Action[AnyContent] =
+    (identify andThen getData(lrn) andThen requireData).async {
+      implicit request =>
+        val preparedForm = request.userAnswers.get(ConfirmRemoveContainerPage(index, containerIndex)) match {
```
Comment (garethgooch) [inline app/controllers/addItems/containers/ConfirmRemoveContainerController.scala:58]: No pre-pop remove

File: test/controllers/addItems/containers/ConfirmRemoveContainerControllerSpec.scala [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+      )
+
+      val jsonWithoutConfig = jsonCaptor.getValue - configKey
+
+      templateCaptor.getValue mustEqual template
+      jsonWithoutConfig mustBe expectedJson
+
+    }
+
+    "must populate the view correctly on a GET when the question has previously been answered" in {
```
Comment (garethgooch) [inline test/controllers/addItems/containers/ConfirmRemoveContainerControllerSpec.scala:92]: no pre-pop remove

**Review by garethgooch — CHANGES_REQUESTED** [review]

File: app/utils/AddItemsCheckYourAnswersHelper.scala (hunk 4 above)
Comment (kevincoatsworth) [reply app/utils/AddItemsCheckYourAnswersHelper.scala:590]: :+1:

File: app/pages/addItems/containers/ConfirmRemoveContainerPage.scala (hunk 1 above)
Comment (kevincoatsworth) [reply app/pages/addItems/containers/ConfirmRemoveContainerPage.scala:1]: :+1:

File: app/controllers/addItems/containers/ConfirmRemoveContainerController.scala (hunk 2 above)
Comment (kevincoatsworth) [reply app/controllers/addItems/containers/ConfirmRemoveContainerController.scala:98]: :+1:

File: app/queries/ContainersQuery.scala (hunk 3 above)
Comment (kevincoatsworth) [reply app/queries/ContainersQuery.scala:23]: :+1:

File: test/pages/addItems/containers/ConfirmRemoveContainerPageSpec.scala (hunk 5 above)
Comment (kevincoatsworth) [reply test/pages/addItems/containers/ConfirmRemoveContainerPageSpec.scala:1]: :+1:

File: test/generators/UserAnswersEntryGenerators.scala (hunk 6 above)
Comment (kevincoatsworth) [reply test/generators/UserAnswersEntryGenerators.scala:36]: :+1:

File: app/controllers/addItems/containers/ConfirmRemoveContainerController.scala (hunk 7 above)
Comment (kevincoatsworth) [reply app/controllers/addItems/containers/ConfirmRemoveContainerController.scala:58]: :+1:

File: test/controllers/addItems/containers/ConfirmRemoveContainerControllerSpec.scala (hunk 8 above)
Comment (kevincoatsworth) [reply test/controllers/addItems/containers/ConfirmRemoveContainerControllerSpec.scala:92]: :+1:

**Pull request merged**